Packages/ClientRuntime/Sources/Waiters/Waiter.swift

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// Makes a waiter operation, as configured by the caller, into an async sequence of waiter status events.
+/// Intended to be a generic type for use when waiting on any Smithy `operation`.
```

Comment:
Nit: 

>Makes a waiter operation, as configured by the caller, into an async sequence of waiter status events.

I think this might be a more appropriate comment for `public func asyncSequence() -> WaiterSequence<Input, Output>` below

>/// Intended to be a generic type for use when waiting on any Smithy `operation`.

I'm struggling a bit to understand the meaning and purpose of this comment. Is it necessary?

Lastly, perhaps we can provide a detailed comment here taking from the Smithy docs

```suggestion
/// A Waiter is a client-side abstraction used to poll a resource until a desired state is reached, or until it is determined that the resource will never enter into the desired state.
```